R/interface.R

```diff
@@ -467,6 +466,38 @@ get.edges <- function(graph, es) {
   ends(graph, es, names = FALSE)
 }
 
+el_to_vec <- function(x, call = rlang::caller_env()) {
+  if (is.data.frame(x)) {
+    if (typeof(x[[1]]) == typeof(x[[2]])) {
+      c(rbind(x[[1]], x[[2]]))
+    } else {
+      cli::cli_abort("The columns of the data.frame are of different type ({typeof(x[[1]])} and {typeof(x[[2]])}) ")
+    }
+  } else if (inherits(x, "matrix")) {
+    dimx <- dim(x)
+    if (identical(dimx, c(2L, 2L))) {
+      lifecycle::deprecate_stop(
+        "2.1.5",
+        "get_edge_ids(vp = 'is not allowed to be a 2 times 2 matrix')"
+      )
+    } else if (dimx[1] == 2L) {
+      lifecycle::deprecate_warn(
+        "2.1.5",
+        "get_edge_ids(vp = 'supplied as a matrix should be a n times 2 matrix, not 2 times n')"
+      )
+      c(x)
+    } else if (dimx[2] == 2L) {
+      c(t(x))
+    } else {
+      cli::cli_abort("{.args vp} was supplied as a {dimx[1]} times {dimx[2]} matrix. Only n times 2 matrices are allowed")
+    }
+  } else if (is.vector(x)) {
+    x
```

Comment:
Ideally, I'd like to soft-deprecate this as well. But this has other dependencies. Let's plan this work and do it later.